### PR TITLE
Use timeout for HTTP POST method to avoid infinite waits

### PR DESCRIPTION
**Pull request by tisnik**
## Type of change

- [ ] Refactor
- [ ] New feature
- [ ] Bug fix
- [x] Optimization
- [ ] Documentation Update

## Description

Use timeout for HTTP POST method to avoid infinite waits

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.

## Testing
- Right now it is not covered by integration tests, unfortunately

### CODE REVIEW

**Comment by tisnik** [comment]
@bparees

**Comment by tisnik** [comment]
@vishnuchalla

File: src/ui/gradio_ui.py [hunk 1]
```diff
@@ -49,7 +51,7 @@ def chat_ui(self, prompt, history, use_history=None):
                     f"Request failed with status code {response.status_code}"
                 )
                 self.logger.info(f"Response text: {response.text}")
-                return "Sorry, an error occurred: " + response.text
+                return f"Sorry, an error occurred: {response.text}"
```
Comment (bparees) [inline src/ui/gradio_ui.py:54]: I don't particularly care either way, but is there a particular reason to prefer this style of string construction?

File: src/ui/gradio_ui.py [hunk 2]
```diff
@@ -36,8 +36,10 @@ def chat_ui(self, prompt, history, use_history=None):
         json_data = json.dumps(data)
 
         try:
-            # Make the HTTP POST request
-            response = requests.post(self.ols_url, headers=headers, data=json_data)
+            # Make the HTTP POST request, wait for response with 10 seconds timeout
+            response = requests.post(
+                self.ols_url, headers=headers, data=json_data, timeout=10
```
Comment (bparees) [inline src/ui/gradio_ui.py:41]: removing my lgtm.  This needs to be higher.  The LLM generation calls can be pretty slow.  Make it 30s.

File: src/ui/gradio_ui.py (hunk 1 above)
Comment (tisnik) [reply src/ui/gradio_ui.py:54]: well Python people find it more idiomatic. TBH I don't care either (returning to Python from Go world), but IMHO we should be consistent

Comment (bparees) [reply src/ui/gradio_ui.py:54]: > IMHO we should be consistent

yeah, i'm also coming from a golang world so i miss the consistent structures the language itself enforces.

I was just curious if there was an actual performance difference or anything.  I'm fine w/ us trying to standardize on use of f-string.

Comment (tisnik) [reply src/ui/gradio_ui.py:54]: no, no performance improvements - Python is still slow, as usual :-)  [and Python 3.11/3.12 is a just bit better: some benchmarks I did https://www.root.cz/obrazek/1110447/]

File: src/ui/gradio_ui.py (hunk 2 above)
Comment (tisnik) [reply src/ui/gradio_ui.py:41]: okay

Comment (tisnik) [reply src/ui/gradio_ui.py]: updated

**Pull request merged**